File: plugin-core/src/main/java/appland/remote/StopRemoteRecordingForm.java
package appland.remote;

import appland.AppMapBundle;
import com.intellij.openapi.util.text.StringUtil;
import com.intellij.ui.TextFieldWithHistory;
import com.intellij.ui.components.JBLabel;
import com.intellij.ui.components.JBTextField;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;
import java.util.List;

public class StopRemoteRecordingForm {
    private JPanel mainPanel;
    private TextFieldWithHistory urlComboBox;
    private JBLabel urlLabel;
    private JBTextField appMapName;
    private JBLabel appMapNameLabel;

    public StopRemoteRecordingForm(@Nullable String activeRecordingUrl, @NotNull List<String> recentRecordingUrls) {
        urlLabel.setText(AppMapBundle.get("appMapRemoteRecording.urlLabel"));
        appMapNameLabel.setText(AppMapBundle.get("appMapRemoteRecording.appMapNameLabel"));

        urlComboBox.setHistory(recentRecordingUrls);
        if (StringUtil.isNotEmpty(activeRecordingUrl)) {
            urlComboBox.setText(activeRecordingUrl);
        }
    }

    public JPanel getMainPanel() {
        return mainPanel;
    }

    @NotNull
    public String getURL() {
        return this.urlComboBox.getText();
    }

    @NotNull
    public String getName() {
        return this.appMapName.getText();
    }

    public JBTextField getAppMapNameInput() {
        return appMapName;
    }

    public TextFieldWithHistory getUrlComboBox() {
        return urlComboBox;
    }
}

File: plugin-core/src/main/kotlin/appland/remote/StopRemoteRecordingForm.kt
package appland.remote

import appland.AppMapBundle
import com.intellij.ui.TextFieldWithHistory
import com.intellij.ui.components.JBTextField
import com.intellij.ui.dsl.builder.AlignX
import com.intellij.ui.dsl.builder.panel

class StopRemoteRecordingForm(activeRecordingUrl: String?, recentUrls: List<String>) {
    lateinit var urlComboBox: TextFieldWithHistory
        private set
    lateinit var appMapNameInput: JBTextField
        private set

    val mainPanel = panel {
        row(AppMapBundle.get("appMapRemoteRecording.urlLabel")) {
            urlComboBox = textFieldWithHistory(recentUrls).align(AlignX.FILL).component
            if (!activeRecordingUrl.isNullOrEmpty()) {
                urlComboBox.text = activeRecordingUrl
            }
        }

        row(AppMapBundle.get("appMapRemoteRecording.appMapNameLabel")) {
            appMapNameInput = textField().align(AlignX.FILL).component
        }
    }

    val url: String
        get() {
            return urlComboBox.text
        }

    val name: String
        get() {
            return appMapNameInput.text
        }
}